test/helper.sh

```diff
@@ -98,72 +102,196 @@ wait_for_managed_cert() {
+    ingress_name="$1"
+    ns="$2"
+    context="$3"
+    fr=$(get_ing_resource "${ingress_name}" "${ns}" "${context}" "ingress.kubernetes.io/https-forwarding-rule")
+    if [[ "${fr}" == "null" ]]; then 
+        fr=$(get_ing_resource "${ingress_name}" "${ns}" "${context}" "ingress.kubernetes.io/forwarding-rule")
+    fi        
+    echo "${fr}"
+}
+
+# Get target http proxy of the given ingress.
+# Arguments:
+#   Name of the ingress.
+#   Namespace of the ingress.
+#   The context where the resource resides.
+# Outputs:
+#   Name of target http proxy for the ingress.
+get_target_http_proxy() {
+    local ingress_name ns context thp
+    ingress_name="$1"
+    ns="$2"
+    context="$3"
+    thp=$(get_ing_resource "${ingress_name}" "${ns}" "${context}" "ingress.kubernetes.io/target-proxy")
+    echo "${thp}"
+}
+
+# Get target https proxy of the given ingress.
+# Arguments:
+#   Name of the ingress.
+#   Namespace of the ingress.
+#   The context where the resource resides.
+#   Name of the target https proxy for the ingress.
+# Outputs:
+#   Name of target https proxy for the ingress.
+get_target_https_proxy() {
+    local ingress_name ns context thsp
+    ingress_name="$1"
+    ns="$2"
+    context="$3"
+    thsp=$(get_ing_resource "${ingress_name}" "${ns}" "${context}" "ingress.kubernetes.io/https-target-proxy")
+    echo "${thsp}"
+}
+
+# Get url map of the given ingress.
 # Arguments:
 #   Name of the ingress.
 #   Namespace of the ingress.
+#   The context where the resource resides.
+# Outputs:
+#   Name of url map for the ingress.
+get_url_map() {
+    local ingress_name ns context um
+    ingress_name="$1"
+    ns="$2"
+    context="$3"
+    um=$(get_ing_resource "${ingress_name}" "${ns}" "${context}" "ingress.kubernetes.io/url-map")
+    echo "${um}"
+}
+
+# Get backend services of the given ingress.
+# Arguments:
+#   Name of the ingress.
+#   Namespace of the ingress.
+#   The context where the resource resides.
+# Outputs:
+#   Name of all backend services for the ingress as a string separated by new line. 
+get_backends() {
+    local ingress_name ns context backend_map
+    ingress_name="$1"
+    ns="$2"
+    context="$3"
+    backend_map=$(get_ing_resource "${ingress_name}" "${ns}" "${context}" "ingress.kubernetes.io/backends")
+    backends=$(echo "${backend_map}" | jq "fromjson | keys[]")
+    echo "${backends[@]}"
+}
+
+# Get the specific resource of the given ingress.
+# Arguments:
+#   Name of the ingress.
+#   Namespace of the ingress.
+#   The context where the resource resides.
+#   Name of the resource in the ingress annotation.
+# Output:
+#   Value of the resource using annotation as key in the given ingress.
+get_ing_resource() {
+    local ingress_name ns context resource_annotation resource
+    ingress_name="$1"
+    ns="$2"
+    context="$3"
+    resource_annotation="$4"
+    resource=$(kubectl get ingress --context="${context}" \
+                                   -n "${ns}" \
```

Comment:
indent like this

kubectl get ingress \
  --context ...   << don't push it all the way to the first param